src/main/java/appeng/parts/p2p/PartP2PIC2Power.java

```diff
@@ -19,247 +19,245 @@
-//		{
-//			this.queueTunnelDrain( PowerUnits.EU, amount );
-//			x.OutputEnergyA = amount;
-//			x.OutputVoltageA = voltage;
-//			return 0;
-//		}
-//
-//		if( x != null && x.OutputEnergyB <= 0.001 )
-//		{
-//			this.queueTunnelDrain( PowerUnits.EU, amount );
-//			x.OutputEnergyB = amount;
-//			x.OutputVoltageB = voltage;
-//			return 0;
-//		}
-//
-//		return amount;
-//	}
-//
-//	public float getPowerDrainPerTick()
-//	{
-//		return 0.5f;
-//	}
-//
-//	@Override
-//	public double getOfferedEnergy()
-//	{
-//		if( this.output )
-//		{
-//			return this.OutputEnergyA;
-//		}
-//		return 0;
-//	}
-//
-//	@Override
-//	public void drawEnergy( double amount )
-//	{
-//		this.OutputEnergyA -= amount;
-//		if( this.OutputEnergyA < 0.001 )
-//		{
-//			this.OutputEnergyA = this.OutputEnergyB;
-//			this.OutputEnergyB = 0;
-//
-//			this.OutputVoltageA = this.OutputVoltageB;
-//			this.OutputVoltageB = 0;
-//		}
-//	}
-//
-//	@Override
-//	public int getSourceTier()
-//	{
-//		if( this.output )
-//		{
-//			return this.calculateTierFromVoltage( this.OutputVoltageA );
-//		}
-//		return 4;
-//	}
-//
-//	private int calculateTierFromVoltage( double voltage )
-//	{
-//		return ic2.api.energy.EnergyNet.instance.getTierFromPower( voltage );
-//	}
-// }
+import java.util.LinkedList;
+import java.util.List;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+
+import ic2.api.energy.tile.IEnergyAcceptor;
+import ic2.api.energy.tile.IEnergyEmitter;
+import ic2.api.energy.tile.IEnergySink;
+import ic2.api.energy.tile.IEnergySource;
+
+import appeng.api.config.PowerUnits;
+import appeng.integration.IntegrationType;
+import appeng.items.parts.PartModels;
+import appeng.me.GridAccessException;
+import appeng.me.cache.helpers.TunnelCollection;
+import appeng.transformer.annotations.Integration.Interface;
+import appeng.transformer.annotations.Integration.InterfaceList;
+import appeng.util.Platform;
+
+
+@InterfaceList( value = {
+		@Interface( iface = "ic2.api.energy.tile.IEnergySink", iname = IntegrationType.IC2 ),
+		@Interface( iface = "ic2.api.energy.tile.IEnergySource", iname = IntegrationType.IC2 )
+} )
+public class PartP2PIC2Power extends PartP2PTunnel<PartP2PIC2Power> implements IEnergySink, IEnergySource
+{
+
+	private static final String TAG_BUFFERED_ENERGY_1 = "bufferedEnergy1";
+	private static final String TAG_BUFFERED_ENERGY_2 = "bufferedEnergy2";
+	private static final String TAG_BUFFERED_VOLTAGE_1 = "outputPacket1";
+	private static final String TAG_BUFFERED_VOLTAGE_2 = "outputPacket2";
+
+	private static final P2PModels MODELS = new P2PModels( "part/p2p/p2p_tunnel_ic2" );
+
+	@PartModels
+	public static List<ResourceLocation> getModels()
+	{
+		return MODELS.getModels();
+	}
+
+	// Buffer the energy + voltage for two IC2 ENET packets
+	private double bufferedEnergy1;
+	private double bufferedVoltage1;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this variable to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1213)